### PR TITLE
fix(fsapp): adjust findRoute to handle RouteCollection types

### PR DESCRIPTION
**Pull request by andyschwob**
Two fixes:
- fixes assigning the correct tabAffinity of external routes due to incorrectly appending the path of RouteCollection routes
- add `statusBarStyle` to type ExternalRoute so we can customize the status bar

### CODE REVIEW

File: packages/fsapp/src/beta-app/router/utils.ts [hunk 1]
```diff
@@ -24,7 +24,9 @@ export const resolveRoutes = async ({
   const getChildPath = (route: Route | RouteCollection) => (
     'initialPath' in route ? route.initialPath : route.path
   );
-
+  const getPrefixedPath = (route: Route | RouteCollection, prefixedPath: string) => (
+    'initialPath' in route ? '' : prefixedPath
+  );
```
Comment (wSedlacek) [inline packages/fsapp/src/beta-app/router/utils.ts:29]: We should probably rename these functions.
`get` implies some type of network or storage transaction which isn't hat is happening here.

The `getPrefixedPath()` is only used in one place so we could just inline that or simply put it in a variable.
I am open to suggestions but `getChildPath()` and `getPrefixedPath()` need better names.

**Review by wSedlacek — COMMENTED** [review]
These changes look good. Now that we are starting to get this to be more stable we should start updating some of the badly named functions as we clean things up.

Do you have any ideas for good names for `getChildPath()` or `getPrefixedPath()`?

File: packages/fsapp/src/beta-app/router/types.ts [hunk 2]
```diff
@@ -175,7 +175,10 @@ export type Route =
  */
 export type Routes = readonly (Route | RouteCollection)[];
 
-export type ExternalRoute = Route & { readonly tabAffinity?: string };
+export type ExternalRoute = Route & {
+  readonly tabAffinity?: string;
+  readonly statusBarStyle?: OptionsStatusBar;
```
Comment (wSedlacek) [inline packages/fsapp/src/beta-app/router/types.ts:180]: Is this necessary when `readonly statusBarStyle?: OptionsStatusBar;` exists on `ComponentRoute` which is part of the `Route` union?

I recall you mentioning there was some sort of type error, but what were the conditions of that?
Maybe we can solve that with type guards?

**Review by wSedlacek — APPROVED** [review]
LGTM

**Review by bweissbart — APPROVED** [review]

**Pull request merged**